### ISSUE TITLE
gcp-auth (image pull) secret not refreshed? 

### ISSUE THREAD

**Issue by HarryWeppner** [GoogleContainerTools/gcp-auth-webhook#105]
We are attempting to use the `gcp-auth` minikube addon and the secret `gcp-auth` to pull images from a Google Artifact Registry.

This works well initially, but it doesn't seem to refresh tokens automatically once they expire. An identified workaround is to disable and (re)enable the addon to force a refresh of the tokens.

Is this the expected behavior?

**Comment by spowelljr** [comment]
There is no auto refresh that I'm aware of, reenabling is currently the only way to refresh the token. I can envision a binary in minikube that runs on a ticker and auto refreshes after X amount of time

**Comment by HarryWeppner** [comment]
@spowelljr, thanks for clarifying. With "reenabling" you are referring to disabling and enabling the `gcp-auth` addon, correct?

**Comment by spowelljr** [comment]
You shouldn't have to disable, if you're fine with all the pods being recreated you can run `minikube addons enable --refresh` and that should remount the credentials into all the pods. Let me know if that works.

**Comment by HarryWeppner** [comment]
In our situation, we don't need credentials within the pod, but we need an up-to-date image pull secret to pull the image from a Google Artifact Registry.

**Comment by spowelljr** [comment]
Sorry, I just checked the code for the pull secret and if it's already present it won't overwrite it.
```
	// check if gcp-auth secret already exists
	secList, err := secrets.List(context.TODO(), metav1.ListOptions{})
	if err != nil {
		return err
	}
	for _, s := range secList.Items {
		if s.Name == gcpAuth {
			return nil
		}
	}
```

So currently the only to ways to refresh the token is:
```
$ minikube addons disable gcp-auth && minikube addons enable gcp-auth
```

Or:
```
$ kubectl delete secret gcp-auth && kubectl delete pod -l app=gcp-auth -n gcp-auth
```

The limitation is that currently the secret will only be added if the `gcp-auth` secret doesn't already exist. And the logic that generates the new token and adds it to the `gcp-auth` secret only runs on pod start. Adding a ticker to delete the secret and regenerate after X amount of time should be easy to implement (every hour?). Let me try creating a quick POC and building an image you can test.

**Comment by HarryWeppner** [comment]
@spowelljr, that would be awesome - thanks for helping!

**Comment by spowelljr** [comment]
@HarryWeppner Here's the binary download with the refresh:

Linux amd64: https://storage.googleapis.com/minikube-builds/18000/minikube-linux-amd64
macOS amd64: https://storage.googleapis.com/minikube-builds/18000/minikube-darwin-amd64

If you need arm64 just change it in the URL

```
$ curl -L <URL> -o minikube-refresh
$ chown +x minikube-refresh
$ ./minikube-refresh delete
$ ./minikube-refresh start
$ ./minikube-refresh addons enable gcp-auth
```

It's currently set to refresh after an hour, you should be able to check this with:
```
$ kubectl get secrets
NAME       TYPE                      DATA   AGE
gcp-auth   kubernetes.io/dockercfg   1      5m41s
```

After an hour if you check it again the age should be reset, let me know if you run into any issues. To full test it make sure the pulling works beyond the time it normally stops working (24 hours?). If all is good I can get it merged into the minikube release.

**Comment by HarryWeppner** [comment]
@spowelljr, amazing - testing now and will provide feedback by the end of the day.

**Comment by HarryWeppner** [comment]
I ran a first test with `https://storage.googleapis.com/minikube-builds/18000/minikube-linux-amd64` , which worked flawlessly!

1. Enabled the `gcp-auth` add-on (with your minikube version)
2. Created K8s resources as before
3. Verified everything was ready
4. Deleted K8s resources
5. Removed images
6. Waited until the secret was re-created
7. Repeated steps 2 and 3 above (including image pulling with a re-created secret)

I will watch the Minikube releases so we can adopt the next version as quickly as possible - thank you so much!
